components/ui/Dialog.tsx

```diff
@@ -0,0 +1,250 @@
+'use client';
+
+import * as DialogPrimitive from '@radix-ui/react-dialog';
+import { motion } from 'framer-motion';
+import {
+  BookmarkCheck,
+  ChevronDown,
+  Info,
+  OctagonAlert,
+  TriangleAlert,
+  X,
+} from 'lucide-react';
+import * as React from 'react';
+import { tv } from 'tailwind-variants';
+import {
+  DialogVariants,
+  type Dialog as DialogType,
+  type DialogVariant,
+} from '~/lib/dialogs/dialog-schemas';
+import { Button } from './Button';
+
+// following this docs to implement this
+// https://www.tailwind-variants.org/docs/slots#slots-with-variants
+const dialogElements = tv({
+  slots: {
+    overlay: 'fixed inset-0 z-50 bg-rich-black/50',
+    content:
+      'fixed left-[50%] top-[50%] z-50 grid w-full max-w-sm sm:max-w-lg gap-1 border bg-background p-5 shadow-lg duration-200 rounded-md sm:rounded-lg',
+    close:
+      'focus:ring-ring absolute right-4 top-4 rounded-sm opacity-70 ring-offset-background transition-opacity hover:opacity-100 focus:outline-none focus:ring-2 focus:ring-offset-2 disabled:pointer-events-none data-[state=open]:bg-accent data-[state=open]:text-muted-foreground',
+    title: 'flex items-center gap-1.5 text-xl font-bold',
+    contentMain: 'text-foreground',
+    footer: 'flex flex-col-reverse sm:flex-row sm:justify-end gap-2',
+  },
+  variants: {
+    type: {
+      [DialogVariants.Error]: {
+        contentMain: 'text-sm',
+      },
+      [DialogVariants.Prompt]: {
+        // contentMain: '',
+      },
+      [DialogVariants.Confirm]: {
+        // contentMain: '',
+      },
+      [DialogVariants.Info]: {
+        // contentMain: '',
+      },
+    },
+  },
+});
+
+// Main Dialog Component
+type DialogProps = {
+  dialog: DialogType;
+  dialogOrder: number;
+  handleOpenChange: (dialog: DialogType) => void;
+  confirmDialog: (dialog: DialogType) => void;
+  cancelDialog: (dialog: DialogType) => void;
```

Comment:
Why do all these function handlers need the whole dialog passing to them? Don't they just need the ID? Even if it needs more than the ID, shouldn't the rest of the dialog object be available in the store in dialog manager?